### ISSUE TITLE
False positive when using function overload

### ISSUE THREAD

**Issue by Gpx** [gund/eslint-plugin-deprecation#77]
I've encountered a false positive while updating [faker](https://www.npmjs.com/package/@faker-js/faker). The problem is that faker is using TypeScript's function overload to type the signature of some methods. Some overloads are valid, while others are deprecated. Something like this:

```ts
foo(value: number): void;
/**
 * @deprecated use numeric inputs instead
 */
foo(value: string): void;
```

In this case, `foo(1)` is valid while `foo('1')` is deprecated.

I believe ESLint can't correctly handle this use case. Imagine this code:

```ts
foo(value);
```

Is `value` a `number` or a `string`? We can only know if we're at runtime or if we start integrating heavily with TypeScript.

In cases like this, where there's a function overload, and only some of the declarations are deprecated, I suggest this plugin ignores the code altogether.

---

For an example of a real-world failing code see https://github.com/faker-js/faker/blob/293f12741acb188fd8c7d9d6b3dc2dd9b52d4214/src/modules/date/index.ts#L70-L190

**Comment by gund** [comment]
I will close this as currently there are no reasons to change how this works.
